### DISCUSSION TITLE
Return validation error on save/edit

### DISCUSSION THREAD

**Discussion by artemklushin** [orchidsoftware/platform#1638] [Q&A] [answered]
Hello!

How can I return error when save or edit data on Screen?

For example we have save method:

```php
public function save(User $user, Request $request)
    {
        $request->validate([
            'user.email' => [
                'required',
                Rule::unique(User::class, 'email')->ignore($user),
            ],
        ]);

        $userData = $request->get('user');
   
        // register in remote service
       
        $user
            ->fill($userData)
            ->save();


        Toast::info(__('User was saved.'));

        return redirect()->route('platform.systems.users');
    }
```

For example I need to save user in remote server and remote server can return error.
I can set Toast::info(__('Failed to save.')); and redirect back to UserEditScreen, but how to set errors state for some fields?

**Comment by tabuna** [comment, marked as answer]
Hi @artemklushin. Form validation, as well as messages, rely on the error mechanism in laravel. 
We have nothing special about this. Example:

```php
return response()->back()->withErrors(['name' => 'The name is required']);
```
